hbase-endpoint/src/main/java/org/apache/hadoop/hbase/coprocessor/AggregateImplementation.java

```diff
@@ -264,7 +279,7 @@ public void getRowNum(RpcController controller, AggregateRequest request,
     List<Cell> results = new ArrayList<>();
     InternalScanner scanner = null;
     PartialResultContext partialResultContext = new PartialResultContext();
-    boolean hasMoreRows = false;
+    boolean hasMoreRows = true;
```

Comment:
This is half of the key fix here. Assume there are more rows to be found until a scanner tells us otherwise.